### ISSUE TITLE
C++11 requirement?

### ISSUE THREAD

**Issue by erwincoumans** [NVIDIAGameWorks/PhysX#87]
So far, the PhysX 4 SDK could be compiled with C++03, but there is one new introduction of C++11:
(I think Adam wrote this FeatherstoneArticulation::getDenseJacobian)
~~~~~~~~~~~
src/PhysX/physx/source/lowleveldynamics/src/DyFeatherstoneArticulation.cpp:3892:3: warning: 'auto' type specifier is a C++11
      extension [-Wc++11-extensions]
                auto jacobian = [&](PxU32 row, PxU32 col) -> PxReal &  { return cache.denseJacobian[nCols * row + col]; } ;
                ^
src/PhysX/physx/source/lowleveldynamics/src/DyFeatherstoneArticulation.cpp:3892:19: error: expected expression
                auto jacobian = [&](PxU32 row, PxU32 col) -> PxReal &  { return cache.denseJacobian[nCols * row + col]; } ;
~~~~~~~~~~~
Is this intentional?

It is easy to fix by adding a function 'jacobian' and passing in the cache and nCols:
~~~~~~~~~~
       PxReal& jacobian(PxArticulationCache& cache, PxU32 nCols, PxU32 row, PxU32 col)
       {
               return cache.denseJacobian[nCols * row + col];
       }
~~~~~~~~~~

I ran into another compile issue with g_WritebackTGSMethods and g_SolveConcludeTGSMethods not using the assignment operator, also easy to fix:
~~~~~~~~~~~~
--- a/src/PhysX/physx/source/lowleveldynamics/src/DyTGSDynamics.cpp
+++ b/src/PhysX/physx/source/lowleveldynamics/src/DyTGSDynamics.cpp
@@ -1331,7 +1331,7 @@ void writeBackContact4(const PxConstraintBatchHeader& hdr, const PxSolverConstra
 
 void writeBack1D4(const PxConstraintBatchHeader& hdr, const PxSolverConstraintDesc* PX_RESTRICT desc, SolverContext* cache);
 
-TGSWriteBackMethod g_WritebackTGSMethods[]
+TGSWriteBackMethod g_WritebackTGSMethods[]=
 {
        0,
        writeBackContact,                                                                                                       // DY_SC_TYPE_RB_CONTACT
@@ -1363,7 +1363,7 @@ void solveConcludeContactExtBlock(const PxConstraintBatchHeader& hdr, const PxSo
 void solveConclude1DBlockExt(const PxConstraintBatchHeader& hdr, const PxSolverConstraintDesc* desc,
        const PxTGSSolverBodyTxInertia* const txInertias, const PxReal elapsedTime, SolverContext& cache);
 
-TGSSolveConcludeMethod g_SolveConcludeTGSMethods[]
+TGSSolveConcludeMethod g_SolveConcludeTGSMethods[]=
 {
        0,
        solveConcludeContactBlock,  
~~~~~~~~~~~~

A fix in my [fork is here](https://github.com/erwincoumans/PhysX/commit/df3278fdf629a9de184b9e45181d7936814466c1).

**Comment by erwincoumans** [comment]
Report in github, reply in twitter: https://twitter.com/PierreTerdiman/status/1158867541013467136
Pierre Terdiman wrote:
~~~~~~~~~~~~~~
We don't mind using c++11 (e.g. other Nvidia projects embrace it big time) but there is no need for it in PhysX at the moment.

The code you reported will most likely be changed in next release.
~~~~~~~~~~~~~~

**Comment by erwincoumans** [comment]
Fixed in 4.1.1, thanks Pierre!
~~~~~~~~~~~

//		auto jacobian = [&](PxU32 row, PxU32 col) -> PxReal &  { return cache.denseJacobian[nCols * row + col]; } ;
		#define jacobian(row, col)	cache.denseJacobian[nCols * (row) + (col)]
		
~~~~~~~~~~~
